strawberryfields/ops.py

```diff
@@ -1823,21 +1817,21 @@ def __init__(self, V, r=None, decomp=True, tol=1e-6):
 
         if decomp:
             th, self.S = williamson(V, tol=tol)
-            self.pure = np.abs(np.linalg.det(V) - (self.hbar/2)**(2*self.ns)) < tol
-            self.nbar = np.diag(th)[:self.ns]/self.hbar - 0.5
-
-        super().__init__([V, r])
+            self.pure = np.abs(np.linalg.det(V) - 1.0) < tol
+            self.nbar = 0.5 * (np.diag(th)[:self.ns] - 1.0)
 
-        self.decomp = decomp
+        super().__init__([V, r])  # V is hbar-independent, r is not
+        self.decomp = decomp  #: bool: if False, use the backend API call instead of decomposition
 
         # FIXME merge() probably does not work for Gaussians if r is not zero?
 
     def _apply(self, reg, backend, **kwargs):
         p = _unwrap(self.p)
         s = sqrt(sf.hbar / 2)  # scaling factor, since the backend API call is hbar-independent
-        backend.prepare_gaussian_state(p[1]/s, p[0]/(s*s), reg)
+        backend.prepare_gaussian_state(p[1]/s, p[0], reg)
 
     def _decompose(self, reg):
+        # FIXME this is a Preparation, are some of the decompositions below missing Vacuum preparations?
```

Comment:
Good catch - I think this can be fixed simply by changing the `Sgate`'s to `Squeezed` state preparations

Reply:
Ok, I'll fix it.

Reply:
Had to add Vacuum preparations as else-cases.